SupportClient/SupportClient/Controllers/ViewController.swift

```diff
@@ -0,0 +1,201 @@
+        newConversationButton.titleLabel?.font = UIFont._17RobotoMedium
+        newConversationButton.layer.cornerRadius = 22
+        view.addSubview(newConversationButton)
+    }
+    
+    func setupFeedbackTableView() {
+        feedbackTableView.delegate = self
+        feedbackTableView.dataSource = self
+        feedbackTableView.tableFooterView = UIView()
+        feedbackTableView.register(FeedbackTableViewCell.self, forCellReuseIdentifier: FeedbackTableViewCell.reuseID)
+        view.addSubview(feedbackTableView)
+    }
+    
+    func setupConstraints() {
+        if !feedbackData.isEmpty {
+            feedbackTableView.snp.makeConstraints { make  in
+                make.leading.trailing.top.bottom.equalToSuperview()
+            }
+        } else {
+            newConversationButton.snp.makeConstraints { make in
+                make.centerX.equalToSuperview()
+                make.bottom.equalToSuperview().inset(50)
+                make.width.equalTo(view.frame.width / 2)
+                make.height.equalTo(45)
+            }
+            feedbackLabel.snp.makeConstraints { make in
+                make.centerX.equalToSuperview()
+                make.bottom.equalToSuperview().inset(view.frame.height / 2)
+            }
+        }
+    }
+    
+    @objc func handleNavigationBarRightTap() {
+        let isEvenNumTaps = countEditTaps % 2 == 0
+        feedbackTableView.setEditing(isEvenNumTaps, animated: true)
+        countEditTaps += 1
+    }
+    
+}
+
+extension ViewController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 75
+    }
+    
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let _ = feedbackTableView.cellForRow(at: indexPath) as! FeedbackTableViewCell
+        var _ = feedbackData[indexPath.row]
+        // TODO: navigate to the messaging page
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            feedbackData.remove(at: indexPath.row)
+            feedbackTableView.deleteRows(at: [indexPath], with: .fade)
+        }
+        // If all conversations removed, remove UITableView and display default screen
+        if feedbackData.isEmpty {
+            feedbackTableView.snp.removeConstraints()
+            newConversationButton.snp.remakeConstraints { make in
+                make.centerX.equalToSuperview()
+                make.bottom.equalToSuperview().inset(50)
+                make.width.equalTo(view.frame.width / 2)
+                make.height.equalTo(45)
+            }
+            feedbackLabel.snp.remakeConstraints { make in
+                make.centerX.equalToSuperview()
+                make.bottom.equalToSuperview().inset(view.frame.height / 2)
+            }
```

Comment:
Same here. Would save us a lot of this remaking constraints, which tends to lead to a lot of weird potential bugs down the line when this logic gets more complex